tests/foreman/cli/test_acs.py

```diff
@@ -0,0 +1,277 @@
+from robottelo.constants.repos import PULP_FIXTURE_ROOT
+
+ACS_UPDATED = 'Alternate Content Source updated.'
+ACS_DELETED = 'Alternate Content Source deleted.'
+ACS_NOT_FOUND = 'Could not find alternate_content_source resource with id'
+
+VAL_FAILED = 'Validation failed'
+VAL_CANT_BLANK = 'can\'t be blank'
+VAL_MUST_BLANK = 'must be blank'
+VAL_CANNOT_BE = 'cannot be'
+
+
+@pytest.mark.tier2
+@pytest.fixture(scope='module')
+def module_synced_repos(module_org):
+    file_product = make_product({'organization-id': module_org.id})
+    file_repo = make_repository(
+        {'product-id': file_product['id'], 'content-type': 'file', 'url': CUSTOM_FILE_REPO}
+    )
+    Repository.synchronize({'id': file_repo['id']})
+    yum_product = make_product({'organization-id': module_org.id})
+    yum_repo = make_repository(
+        {'product-id': yum_product['id'], 'content-type': 'yum', 'url': CUSTOM_RPM_REPO}
+    )
+    Repository.synchronize({'id': yum_repo['id']})
+    return {'yum': yum_repo, 'file': file_repo}
+
+
+@pytest.mark.tier2
+@pytest.mark.parametrize('cnt_type', ['yum', 'file'])
+@pytest.mark.parametrize('acs_type', ['custom', 'simplified', 'rhui'])
+def test_positive_CRUD_all_types(request, target_sat, acs_type, cnt_type, module_synced_repos):
+    """Create, read, update, refresh and delete ACSes of all supported types.
+
+    :id: 3d51b810-f81f-4d09-80a1-4f27fda4e84f
+
+    :parametrized: yes
+
+    :steps:
+        1. Create an ACS using necessary params.
+        2. List it, read it.
+        3. Update name and description.
+        4. Refresh it.
+        5. Delete it.
+
+    :expectedresults:
+        1. ACS can be created.
+        2. ACS can be listed and read with correct values.
+        3. ACS can be updated and read with new name.
+        4. ACS can be refreshed.
+        5. ACS can be deleted.
+
+    """
+    if 'rhui' in request.node.name and 'file' in request.node.name:
+        pytest.skip('unsupported parametrize combination')
+
+    # Create
+    paths = {'yum': 'rpm-zchunk/,rpm-modular/', 'file': 'file-large/,file-many/'}
+    params = {
+        'name': gen_alphanumeric(),
+        'alternate-content-source-type': acs_type,
+        'content-type': cnt_type,
+        'smart-proxy-ids': target_sat.nailgun_capsule.id,
+    }
+
+    if acs_type == 'simplified':
+        params.update(
+            {
+                'product-ids': module_synced_repos[cnt_type]['product']['id'],
+            }
+        )
+    else:
+        params.update(
+            {
+                'base-url': PULP_FIXTURE_ROOT,
+                'subpaths': paths[cnt_type],
+                'verify-ssl': 'false',
+            }
+        )
+
+    new_acs = target_sat.cli.ACS.create(params)
+
+    # List
+    list = target_sat.cli.ACS.list({'search': f'name={new_acs["name"]}'})
+    assert len(list) == 1
+    assert list[0]['id'] == new_acs['id']
+    assert list[0]['type'] == new_acs['alternate-content-source-type']
```

Comment:
non-blocking, having an assertion messages present helps better readability and possibly TFA